sns/parity_math.c

```diff
@@ -500,29 +562,93 @@ static void reed_solomon_encode(struct m0_parity_math *math,
 
 #undef PARITY_MATH_REGION_ENABLE
 }
+#endif /* __KERNEL__ */
+
+#ifndef __KERNEL__
+static void isal_encode(struct m0_parity_math *math,
+			const struct m0_buf *data,
+			struct m0_buf *parity)
+{
+	uint32_t  i;
+	uint32_t  block_size;
+	int	  ret = 0;
+
+	M0_PRE(math != NULL);
+	M0_PRE(data != NULL);
+	M0_PRE(parity != NULL);
+	M0_PRE(math->pmi_data_count >= 1);
+	M0_PRE(math->pmi_parity_count >= 1);
+	M0_PRE(math->pmi_data_count >= math->pmi_parity_count);
+	M0_PRE(math->pmi_data_count <= SNS_PARITY_MATH_DATA_BLOCKS_MAX);
+
+	uint8_t  *frags_in[math->pmi_data_count];
+	uint8_t  *frags_out[math->pmi_parity_count];
```

Comment:
Variable-sized array object. Let's ask @nikitadanilov to review this code block.